### ISSUE TITLE
CERN contrib: roles not updated when logging in with a token

### ISSUE THREAD

**Issue by ntarocco** [inveniosoftware/invenio-oauthclient#291]
Package version: 1.5.1

## Describe the bug
[This change](https://github.com/inveniosoftware/invenio-oauthclient/issues/228) introduced a new issue that users roles are not fetched when logging in using a token.

The check for the login method via token should happen before checking if the login was done using oauth flow.
